contracts/token/CRC1155/extensions/CRC1155Enumerable.sol

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "./ICRC1155Enumerable.sol";
+import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+
+abstract contract CRC1155Enumerable is ERC1155, ICRC1155Enumerable {
+    using EnumerableSet for EnumerableSet.UintSet;
+
+    // All token ids for enumeration.
+    EnumerableSet.UintSet private _allTokens;
+
+    // Mapping from token ID to token total supply.
+    mapping(uint256 => uint256) private _totalSupplies;
+
+    // Mapping from owner to owned token IDs for enumeration.
+    mapping(address => EnumerableSet.UintSet) private _ownedTokens;
+
+    /**
+     * @dev Returns the total amount of tokens stored by the contract.
+     */
+    function totalSupply() public view virtual override returns (uint256) {
+        return _allTokens.length();
```

Comment:
This method should return _allTokens.length or the sum of every token amount.
Due to pre PR's discussion, this should be the second one right?

Reply:
No, `totalSupply()` returns the total amount of token IDs in CRC1155 contract. It could be used to enumerate all the token IDs in contract, especially for a pure NFT contract.